py3status/modules/keyboard_layout.py

```diff
@@ -2,88 +2,97 @@
 """
 Display the current keyboard layout.
 
-Configuration parameters:
-    - cache_timeout: check for keyboard layout change every seconds
-
+This module displays the current active keyboard layout.
 Requires:
     - xkblayout-state
     or
-    - setxkbmap
+    - setxkbmap and xset (Works for the first two predefined layouts.)
 
-@author shadowprince
+Configuration parameters:
+    - cache_timeout: check for keyboard layout change every seconds
+    - colors: a comma separated string of color values for each layout,
+              eg: "us=#FCE94F, fr=#729FCF".
+    - color: a single color value for all layouts. eg: "#FCE94F"
+
+@author shadowprince, tuxitop
 @license Eclipse Public License
 """
 
 from subprocess import check_output
 from time import time
-import shlex
 import re
 
-# colors of layouts, check your command's output to match keys
-LANG_COLORS = {
-    'fr': '#268BD2',  # solarized blue
-    'ru': '#F75252',  # red
-    'ua': '#FCE94F',  # yellow
-    'us': '#729FCF',  # light blue
-}
-
-LAYOUT_RE = re.compile(r".*layout:\s*(\w+).*", flags=re.DOTALL)
-
-
-def xkblayout():
-    """
-    check using xkblayout-state (preferred method)
-    """
-    return check_output(
-        ["xkblayout-state", "print", "%s"]
-    ).decode('utf-8')
-
-
-def setxkbmap():
-    """
-    check using setxkbmap >= 1.3.0
-
-    Please read issue 33 for more information :
-        https://github.com/ultrabug/py3status/pull/33
-    """
-    out = check_output(shlex.split("setxkbmap -query")).decode("utf-8")
-
-    return re.match(LAYOUT_RE, out).group(1)
-
 
 class Py3status:
     """
     """
     # available configuration parameters
-    cache_timeout = 10
-    color = ''
+    cache_timeout = 1
```

Comment:
please do not change default cache timeouts, running a command every second is kinda overkill, leave it to 10s